src/de/mod.rs

```diff
@@ -104,21 +104,86 @@ impl<'de> Deserializer<'de> {
         // Create a working copy
         let mut bytes = self.bytes;
 
-        if bytes.consume("(") {
-            bytes.skip_ws()?;
+        // The caller checks for a brace before calling this
+        if !bytes.consume("(") {
```

Comment:
If it's unreachable, why are we checking for it?
If we aren't sure, we should put an assert in. If we are sure, just don't check.

Reply:
Fixed it. There is probably a lot more that can be made more beautiful.

My changes are basically just a proof for myself that fixing this is possible but it ended up working so I'm already using this code myself.